iothub/device/src/ClientFactory.cs

```diff
@@ -82,13 +98,19 @@ public static InternalClient Create(string hostname, string gatewayHostname, IAu
 
             if (transportType != TransportType.Amqp_Tcp_Only
                 && transportType != TransportType.Mqtt_Tcp_Only
-                && authenticationMethod is DeviceAuthenticationWithX509Certificate
-                && ((DeviceAuthenticationWithX509Certificate)authenticationMethod).ChainCertificates != null)
+                && authenticationMethod is DeviceAuthenticationWithX509Certificate certificate
+                && certificate.ChainCertificates != null)
             {
                 throw new ArgumentException("Certificate chains are only supported on Amqp_Tcp_Only and Mqtt_Tcp_Only");
             }
 
-            IotHubConnectionStringBuilder connectionStringBuilder = IotHubConnectionStringBuilder.Create(hostname, gatewayHostname, authenticationMethod);
+            if (!string.IsNullOrWhiteSpace(options?.ModelId)
+                && transportType == TransportType.Http1)
+            {
+                throw new InvalidOperationException("PnP is not supported over the HTTP transport.");
```

Comment:
Abbreviations may cause confusion here. How about "Plug and play is not supported over HTTP"?

Reply:
Changed